snapcraft/cli/snapcraftctl/_runner.py

```diff
@@ -72,13 +71,25 @@ def prime():
 @click.argument("version")
 def set_version(version):
     """Set the version of the snap"""
+
+    if not version:
```

Comment:
How about using click's native features for this https://click.palletsprojects.com/en/7.x/options/#callbacks-for-validation ?

Reply:
Certainly can.  and makes sense here, will do.